File: app/src/main/java/org/dicio/dicio_android/util/RegexUtils.java
package org.dicio.dicio_android.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public final class RegexUtils {

    private RegexUtils() {
    }

    // taken from NewPipeExtractor, file utils/Parser.java, created by @theScrabi
    public static String matchGroup1(final Pattern pattern, final String string) {
        return matchGroup(pattern, string, 1);
    }

    // taken from NewPipeExtractor, file utils/Parser.java, created by @theScrabi
    public static String matchGroup(final Pattern pattern, final String string, final int group) {
        final Matcher matcher = pattern.matcher(string);
        final boolean foundMatch = matcher.find();
        if (foundMatch) {
            return matcher.group(group);
        } else {
            // only pass input to exception message when it is not too long
            if (string.length() > 1024) {
                throw new IllegalArgumentException("failed to find pattern \"" + pattern.pattern());
            } else {
                throw new IllegalArgumentException("failed to find pattern \"" + pattern.pattern()
                        + " inside of " + string + "\"");
            }
        }
    }

    public static String replaceAll(final Pattern pattern,
                                    final String string,
                                    final String replacement) {
        return pattern.matcher(string).replaceAll(replacement);
    }
}

File: app/src/main/java/org/stypox/dicio/util/RegexUtils.kt
package org.stypox.dicio.util

import java.util.regex.Pattern

object RegexUtils {
    // taken from NewPipeExtractor, file utils/Parser.java, created by @theScrabi
    fun matchGroup(pattern: Pattern, string: String, group: Int = 1): String {
        val matcher = pattern.matcher(string)
        val foundMatch = matcher.find()
        return if (foundMatch) {
            matcher.group(group)!!
        } else {
            // only pass input to exception message when it is not too long
            if (string.length > 1024) {
                throw IllegalArgumentException("failed to find pattern \"" + pattern.pattern())
            } else {
                throw IllegalArgumentException(
                    "failed to find pattern \"" + pattern.pattern()
                            + " inside of " + string + "\""
                )
            }
        }
    }

    fun replaceAll(
        pattern: Pattern,
        string: String,
        replacement: String
    ): String {
        return pattern.matcher(string).replaceAll(replacement)
    }
}